### DISCUSSION TITLE
作者你好，Monorepo 有个相关问题想请教你，望告知

### DISCUSSION THREAD

**Discussion by cbombonera** [vbenjs/vue-vben-admin#4657] [General]
本人对前端了解本来就不足，看到Monorepo 很吸引我，也查阅了些资料
分层很像做后台一样，很有意思

我的问题如下

我看作者的packages 封装的是一些公共组件，钩子等等  apps 使用这些公共的
然后 apps层 有 对应的 ele antd ...

比如我是用**element plus** 做开发，那么我的业务就写在**web-ele**层下
但是我在思考一个问题，我可能也会封装一些 element plus的组件 这里主要说到的不是业务组件，主要是优化组件
比如对element plus 大多组件再封装拓展的组件，比如 XxElMenu，XxElTabl......

1.按传统的思维 在**web-ele** 目录下建立 components文件夹存放是否Monorepo 的设计理念?
2.如果也想和作者一样，把它变成包，放在哪个目录上比较合适呢，我看放到 **packages**层应该是不合适的，毕竟是公用的，比如web-ant就不需要这个包 也就是如果有个ant组开发 他是并不会用到它的，我想着在  apps层 建立一个web-ele-components【工作区】  然后只是web-ele引用它，不知道在分层上合不合适？

**Comment by anncwb** [comment]
packages 除了 @core 目录外，理论上可以存放你们多个app的业务代码，是可以更改的，这些包大部分都是对core的简单转发，同时这些代码没有耦合ui库，所以可以复用，如果你们想划分一个包出来，可以建一个packages/business（名字自己决定）, 用于存放业务代码、复用、组件等等，至于里面怎么分包你们自己考量了

**Reply by cbombonera** [reply]
> packages 除了 @core 目录外，理论上可以存放你们多个app的业务代码，是可以更改的，这些包大部分都是对core的简单转发，同时这些代码没有耦合ui库，所以可以复用，如果你们想划分一个包出来，可以建一个packages/business（名字自己决定）, 用于存放业务代码、复用、组件等等，至于里面怎么分包你们自己考量了

谢谢

**Reply by cbombonera** [reply]
> packages 除了 @core 目录外，理论上可以存放你们多个app的业务代码，是可以更改的，这些包大部分都是对core的简单转发，同时这些代码没有耦合ui库，所以可以复用，如果你们想划分一个包出来，可以建一个packages/business（名字自己决定）, 用于存放业务代码、复用、组件等等，至于里面怎么分包你们自己考量了

还有个问题想咨询一下
比如我建立了一个packages/ep
只有**web-ele** 引用了它， **web-antd** 没有引用它
那么在**web-antd** 打包部署后，应该是没有packages/ep 相关的内容吧

**Comment by anncwb** [comment]
没有的
